src/runtime_src/driver/xclng/tools/xbutil/xbutil.h

```diff
@@ -700,6 +701,30 @@ class device {
         }
         parseComputeUnits( computeUnits );
 
+        /**
+         * \note Adding device information for debug and profile
+         * This will put one more section debug_profile into the
+         * json dump that shows all the device information that
+         * debug and profile code in external systems will need
+         * e.g. sdx_server, hardware_sercer, GUI, etc
+         */
+        xclDebugProfileDeviceInfo info;
+        int err = xclGetDebugProfileDeviceInfo(m_handle, &info);
+        unsigned int device_index = (unsigned int)info.device_index;
+        unsigned int user_instance = (unsigned int)info.user_instance;
+        unsigned int mgmt_instance = (unsigned int)info.mgmt_instance;
+        unsigned int nifd_instance = (unsigned int)info.nifd_instance;
+        std::string device_name = (std::string)info.device_name;
+        std::string nifd_name = (std::string)info.nifd_name;
```

Comment:
Let's get rid of all explicit casts.  In this case they are not necessary.  If a cast is necessary, the use proper C++ conversion casts, e.g. static_cast.  Is there any reason to save the info data in a variable?

Reply:
Just pushed a new one to remove the complexity and resolve conflict with newly added p2p